<launch>
	<!-- Set up arguments for foreground single and multi agent cases -->
        <arg name="open_kernel_size" default="3" />
        <arg name="median_kernel" default="3" />
        <arg name="fg_depth_thresh" default="1500" />

    <node pkg="tof_preprocessing"
          type="preprocessing.py"
          name="depth_preprocessing"
          output="screen">

        <param name="min_depth" value="150" />
        <param name="max_depth" value="6000" />
        <param name="bg_frames" value="50" />
	<param name="bg_beta" value="0.999" />
	<param name="bg_beta2" value="0.9" />
	<param name="fg_threshold" value="200" />

	<param name="fg_age_threshold" value="50" />
	<param name="static_scene_thresh" value="30" />

	<param name="min_blob_area" value="3000" />
	<param name="min_blob_size" value="300" />
        <param name="track_dist_thresh" value="50" />
        <param name="track_timeout" value="15" />
	<param name="min_depth_values" value="100" />
	<param name="depth_variation_thresh" value="300" />
	<param name="peaks_threshold" value="0.4" />

	<param name="fg_blob_mem" value="60" />
	<param name="fg_bbox_mem" value="20" />
	<param name="fg_motion_threshold" value="1.5" />
	<param name="fg_bbox_motion_threshold" value="500" />

	<param name="fg_depth_thresh" value="$(arg fg_depth_thresh)" />
	<param name="median_kernel" value="$(arg median_kernel)" />
	<param name="open_kernel_size" value="$(arg open_kernel_size)"/>

    </node>
</launch>
